libsolidity/formal/SymbolicState.h

```diff
@@ -191,13 +193,15 @@ class SymbolicState
 		m_context
 	};
 
+	// TODO: Remove `block.difficulty` in 0.9.0
 	BlockchainVariable m_tx{
 		"tx",
 		{
 			{"block.basefee", smtutil::SortProvider::uintSort},
 			{"block.chainid", smtutil::SortProvider::uintSort},
 			{"block.coinbase", smt::smtSort(*TypeProvider::address())},
 			{"block.difficulty", smtutil::SortProvider::uintSort},
+			{"block.prevrandao", smtutil::SortProvider::uintSort},
```

Comment:
Because we still accepting `block.difficulty` until version `0.9.0`. We can completely replace it by `block.prevrandao` in the `0.9.0` release.

Reply:
But this code is not responsible for accepting it, is it? It's keeping track of the state of these variables and both `difficulty` and `prevrandao` are internally the same exact piece of state. It seems to me that we could remove it already if we translated both to `prevrandao`.

Reply:
@cameel we could do that, but since `block.difficulty` is still part of the language, we would have to change the code in https://github.com/ethereum/solidity/blob/develop/libsolidity/formal/SMTEncoder.cpp#L1340 from being generic over what's being translated to an extra (and hacky imo) `if difficulty use prevrandao`. The current version adds `prevrandao` as an extra symbolic variable (by being added in the map above) and adds constraints that those need to be the same (which is also somewhat hacky). So IMO either is fine. But just to clarify the options:

1. Declare (in SMT) both `difficulty` and `prevrandao`, add constraints that they're the same in SMT (current version).
2. Keep only `prevrandao` in the SMT map above, and change the linked SMTEncoder.cpp (or the internal access in SymbolicState.cpp) adding a special case for `block.difficulty` which translates directly to `block.prevrandao` in SMT.

Reply:
ok, if both options are hacky then I'm fine with staying with this. Just thought we could get rid of the TODO immediately. instead of having to wait for 0.9.0.

Reply:
actually now I think the other option is better... I'll quickly do it and push it directly here with updated tests

Reply:
we should also have an SMT test that asserts that prevrandao and difficulty are the same. I'll add it in my commit too